### DISCUSSION TITLE
404 on a page when deployed but not in localhost

### DISCUSSION THREAD

**Discussion by astrosiac** [vercel/community#1766] [Help] [answered]
### Summary

I am having an issue with my Next.js and firebase v9 app. When I deploy my app I have a page called pages/teams/[teamId].jsx. When a user creates a new team on the readers-hub.js page the doc is added to the team collection just fine. But when they try to navigate to the page it gives a firebase 404 error. NOW if I re-redeploy the site using vercel and then the user navigates back to the [teamId] url the page loads and does not give a 404 error. This is also not the case with localhost. But I do get the same issue with a local production environment.

### Example

rqreadathon.org

**Comment by amyegan** [comment, marked as answer]
Hi @astrosiac! Is your site SSR or SSG? If it's a static site, you'll need to use [ISR](https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration) to signal that the new [teamId] page should be generated on first visit or when the new team is created.

**Reply by astrosiac** [reply]
My site is I believe static
